libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5173,6 +5173,32 @@ MAV_RESULT GCS_MAVLINK::handle_do_set_safety_switch_state(const mavlink_command_
     }
 }
 
+#if AP_MAVLINK_MAV_CMD_AWR_SYS_CMP_ID_ENABLED
+MAV_RESULT GCS_MAVLINK::handle_do_set_sys_cmp_id(const mavlink_command_int_t &packet, const mavlink_message_t &msg)
+{
+    const uint8_t new_system_id = packet.param1;
+    const uint8_t new_component_id = packet.param2;
+    const uint8_t do_reboot = packet.param3;
+
+    if (new_system_id == 0 || hal.util->get_soft_armed()) {
```

Comment:
I wonder why we need to reject the command if `new_system_id` is zero (apart from "that's what the message says")

This means you can't tell a system to *just* change its component ID.

... but you *can* tell it to just change its system ID.  This is weird.